tests/testthat/test_PlainTextParam.R

```diff
@@ -12,42 +12,80 @@ test_that("storeResults,loadResults,PlainTextParam,MsBackendMzR works", {
     ## Loading data again
     b2 <- loadResults(object = MsBackendMzR(), param)
     expect_true(inherits(b2, "MsBackendMzR"))
-    b <- dropNaSpectraVariables(b) #the function does this to be robust, is it a problem ? i should mention it in the doc
+    b <- dropNaSpectraVariables(b)
     expect_equal(b@spectraData, b2@spectraData)
-    expect_equal(peaksVariables(b), peaksVariables(b2))  # true even without forcing the slot
+    expect_equal(peaksVariables(b), peaksVariables(b2))
+    expect_equal(mz(b[1:20]), mz(b2[1:20]))
+
+    ## Check the spectraPath parameter.
+    bp <- dataStorageBasePath(b)
+    ## manually change dataStorage path of backend
+    sd <- read.table(file.path(param@path, "backend_data.txt"), header = TRUE)
+    sd$dataStorage <- sub("faahKO", "other", sd$dataStorage)
+    write.table(sd, file = file.path(param@path, "backend_data.txt"),
+                sep = "\t", quote = FALSE, row.names = FALSE)
+    A <- loadResults(MsBackendMzR(), param)
+    expect_error(validObject(A), "invalid class")
+    A <- loadResults(MsBackendMzR(), param, spectraPath = bp)
+    expect_true(validObject(A))
+
+    param <- PlainTextParam(tempdir())
+    expect_error(loadResults(MsBackendMzR(), param), "No 'backend_data")
 })
 
 test_that("storeResults,loadResults,PlainTextParam,Spectra works", {
     pth <- file.path(tempdir(), "test1")
     param <- PlainTextParam(path = pth)
-    #add processingQueueVariables to test export
+    ## add processingQueueVariables to test export
+    s@processingQueueVariables <- c(s@processingQueueVariables, "rtime")
     s <- filterMzRange(s, c(200,300))
+    s <- filterRt(s, c(3000, 3500)) # to ensure subsetted object would work
```

Comment:
this is to ensure that not only the processing queue is correctly exported/imported, but also eventual data subsets.

Reply:
Aaah sorry I misunderstood that